lib/bots/versionbot.ts

```diff
@@ -1550,6 +1571,157 @@ export class VersionBot extends ProcBot {
 	}
 
 	/**
+	 * Loops through all of the commits for a PR, ensuring that any required tags are present the
+	 * specified number of times for a PR.
+	 *
+	 * @param allCommits  All of the commits for the PR.
+	 * @param config      The config (if any) for the repository the PR belongs to.
+	 * @returns           An array of MissingTag objects, denoting required tags not on the PR.
+	 */
+	private checkCommitFooterTags(allCommits: GithubApiTypes.Commit[], config: VersionBotConfiguration): MissingTag[] {
+		const tagDefinitions = (config || {})['required-tags'] || [];
+		let tagCounts: { [key: string]: number } = {};
+		let changeTypeFound = false;
+
+		// Filter the list of commits to those not made by VB.
+		const commits = _.filter(allCommits, (commit) => commit.commit.committer.name !== process.env.VERSIONBOT_NAME);
+
+		// Get and validate tag configs.
+		for (let tag of tagDefinitions) {
+			// Lower the tag name.
+			tag.name = tag.name.toLowerCase();
+
+			if (tagCounts[tag.name]) {
```

Comment:
If the tag names have to be unique, why not have the configuration be a `{ tagName: { config } }` object, rather than a list where `name` is special? I.e. the config could look like:

```yaml
required_tags:
  connects-to:
    occurance: 1
  change-type:
    occurrence: 1
    values: /\s*(patch|minor|major)\s*$/i  
```